firmware/Platypus.cpp

```diff
@@ -347,48 +365,156 @@ Sensor::Sensor(int channel)
   digitalWrite(board::LOOPBACK, LOW);
   
   // Disable 12V output
-  pinMode(board::SENSOR[channel].PWR_ENABLE, OUTPUT);
-  digitalWrite(board::SENSOR[channel].PWR_ENABLE, LOW);
-  
-  // Register serial event handler
-  SerialHandler_t handler = {Sensor::onSerial_, this}; 
-  SERIAL_HANDLERS[channel] = handler;
+  pinMode(board::SENSOR_PORT[port].PWR_ENABLE, OUTPUT);
+  digitalWrite(board::SENSOR_PORT[port].PWR_ENABLE, LOW);
 }
 
-void Sensor::calibrate(int flag){
-  
-}
 
-Sensor::~Sensor()
+AnalogSensor::AnalogSensor(int id, int port)
+  : ExternalSensor(id, port), scale_(1.0f), offset_(0.0f) {}
+
+bool AnalogSensor::set(const char* param, const char* value)
 {
-  // TODO: fill me in
+  // Set analog scale.
+  if (!strncmp("scale", param, 6))
+  {
+    float s = atof(value);
+    scale(s);
+    return true;
+  }
+  // Set analog offset.
+  else if (!strncmp("offset", param, 7))
+  {
+    float o = atof(value);
+    offset(o);
+    return true;
+  }
+  // Return false for unknown command.
+  else
+  {
+    return false;
+  }
 }
 
-bool Sensor::set(const char* param, const char* value)
+void AnalogSensor::scale(float scale)
 {
-  return false;
+  scale_ = scale;
 }
 
-void Sensor::onSerial() 
+void AnalogSensor::offset(float offset)
 {
-  // Default to doing nothing on serial events. 
+  offset = offset_;
```

Comment:
Yea you're right. I'll make the change. We need to do another style cleanup because we have a variable naming mess.